src/Relude/Extra/Enum.hs

```diff
@@ -23,6 +26,21 @@ import Relude
 universe :: (Bounded a, Enum a) => [a]
 universe = [minBound .. maxBound]
 
+{- | Create a function that parses values of type @a@
+
+>>> data Color = Red | Green | Blue deriving (Show, Enum, Bounded)
+>>> parse = makeParser :: String -> Maybe Color
+>>> parse "Red"
+Just Red
+>>> parse "Black"
+Nothing
+-}
+makeParser :: forall a. (Bounded a, Enum a, Show a) => String -> Maybe a
```

Comment:
Isn't it just `readMaybe` function?

Reply:
@vrom911 You're right. @ghallak we don't want to have `Show` constraint. It's better to take custom function from `a`. Also, `String` type is inefficient. It's better to use `Text` or `ByteString`. But we can generalize this even more if you pass function of type `(a -> k)`.

Reply:
Also, name `makeParser` is not precise enough. Better name would be something like `inverseMap`